crates/driver/src/domain/competition/order/mod.rs

```diff
@@ -133,20 +133,57 @@ impl Order {
         matches!(self.kind, Kind::Liquidity)
     }
 
-    /// The buy asset to pass to the solver. This is a special case due to
-    /// orders which buy ETH. The settlement contract only works with ERC20
-    /// tokens, but unfortunately ETH is not an ERC20 token. We still want to
-    /// provide a seamless user experience for ETH trades, so the driver
-    /// will encode the settlement to automatically unwrap the WETH into ETH
-    /// after the trade is done.
+    /// The sell asset to pass to the solver.
     ///
-    /// For this reason, we want the solvers to solve the orders which buy ETH
-    /// as if they were buying WETH, and then add our unwrap interaction to that
-    /// solution.
+    /// In order to simplify solver logic, we scale the remaining sell amount
+    /// for orders that have been partially filled to simplify computations on
+    /// the solver side. This is so the solvers only see "whats left".
+    pub fn solver_sell(&self) -> eth::Asset {
```

Comment:
I was quite puzzled by what `solver_sell` was supposed to mean. Can't this be `remaining_sell/buy` (and I agree with Dusan that we probably have this logic in the codebase already somewhere)